evidence.go

```diff
@@ -0,0 +1,279 @@
+		if err := message.UnmarshalCBOR(*e.message.CcaPlatformToken); err != nil {
+			return fmt.Errorf("failed CBOR decoding for CWT: %w", err)
+		}
+
+		pclaims, err := ptoken.DecodeClaims(message.Payload)
+		if err != nil {
+			return fmt.Errorf("failed CBOR decoding of CCA platform claims: %w", err)
+		}
+		e.CcaPlatformClaims = pclaims
+	} else {
+		if err := message.UnmarshalCBOR(*e.message.CcaRealmToken); err != nil {
+			return fmt.Errorf("failed CBOR decoding for CWT: %w", err)
+		}
+		rclaims, err := DecodeClaims(message.Payload)
+		if err != nil {
+			return fmt.Errorf("failed CBOR decoding of CCA realm claim: %w", err)
+		}
+		e.CcaRealmClaims = rclaims
+	}
+	return nil
+}
+
+func (e *CcaEvidence) Verify(iak crypto.PublicKey, rpk crypto.PublicKey) error {
+	if e.message == nil {
+		return fmt.Errorf("no message found")
+	}
+
+	// Check CCA Platform Token
+	if e.message.CcaPlatformToken == nil {
+		return fmt.Errorf("missing CCA platform Token")
+	}
+
+	// First verify the platform token
+	if err := e.verifyCoseToken(*e.message.CcaPlatformToken, iak); err != nil {
+		return fmt.Errorf("unable to verify platform token: %w", err)
+	}
+
+	// Check CCA Realm Token
+	if e.message.CcaRealmToken == nil {
+		return fmt.Errorf("missing CCA realm Token")
+	}
+
+	// Next verify the realm token
+	if err := e.verifyCoseToken(*e.message.CcaRealmToken, rpk); err != nil {
+		return fmt.Errorf("unable to verify realm token: %w", err)
+	}
+	return nil
+}
+
+func (e *CcaEvidence) verifyCoseToken(token []byte, pk crypto.PublicKey) error {
+
+	message := cose.NewSign1Message()
+	if err := message.UnmarshalCBOR(token); err != nil {
+		return fmt.Errorf("failed CBOR decoding for CWT: %w", err)
+	}
+
+	protected := message.Headers.Protected
+	algo, err := protected.Algorithm()
+	if err != nil {
+		return fmt.Errorf("unable to get verification algorithm: %w", err)
+	}
+
+	verifier, err := cose.NewVerifier(algo, pk)
+	if err != nil {
+		return fmt.Errorf("unable to instantiate verifier: %w", err)
+	}
+
+	err = message.Verify([]byte(""), verifier)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// GetInstanceID returns the InstanceID claim from Platform claims that is to be used to locate the
+// verification key or a nil pointer if no suitable InstanceID could be located.
+// A call to this function on CcaEvidence that has not been successfully claims validated
+// is meaningless.
+func (e *CcaEvidence) GetInstanceID() *[]byte {
+	instID, err := e.CcaPlatformClaims.GetInstID()
+	if err != nil {
+		return nil
+	}
+	return &instID
+}
+
+// GetRealmPublicKey returns the Public Key claim from Realm claims
+// Realm Public Key is used to verify the signature on the Realm Token
+// A call to this function on CcaEvidence that has not been successfully claims validated
+// is meaningless.
+func (e *CcaEvidence) GetRealmPublicKey() *[]byte {
+	pubKey, err := e.CcaRealmClaims.GetPubKey()
+	if err != nil {
+		return nil
+	}
+	return &pubKey
+}
```

Comment:
Don't we need a GetImplID interface too?

Reply:
OK, I just realized we need to be symmetrical. I will add it here and tracked the same inside psatoken using:
https://github.com/veraison/psatoken/issues/40